src/components/CompareResults/ResultsMain.tsx

```diff
@@ -55,6 +103,17 @@ function ResultsMain() {
             <div className={styles.title}>Results</div>
             <div className={styles.content}>
               <SearchInput onChange={setSearchTerm} />
+              {isLoading ? (
+                <Box display='flex' sx={{ margin: 'auto' }}>
+                  <CircularProgress />
+                </Box>
+              ) : (
+                <ResultsFrameworkDropdown
+                  frameworkId={frameworkIdVal}
+                  onChange={onFrameworkChange}
+                />
+              )}
```

Comment:
This switching makes that a keyboard user loses focus, which isn't great for accessibility... so I'm not super happy about that either. I think we should use the global spinner mechanism rather than come up with a new mechanism just here.

And probably that global spinner should use the Await/Suspense...

Reply:
Conditional rendering is such a normal workflow in react; I'm sure there's a way to address the accessibility weakness here for losing focus. However, let me go ahead and test whether removing defer and caching the results will show the spinner for same page updates _without having to reload the page._

Reply:
Like I said above, I'd be happy removing this once I cache the revisions and remove defer, thus allowing the spinner to appear on same page updates.